src/app/components/search/search.component.ts

```diff
@@ -1,30 +1,71 @@
-import { AfterViewChecked, Component } from '@angular/core';
-import { initAll } from '@scottish-government/pattern-library/src/all';
+import { CourthouseComponent } from './../common/courthouse/courthouse.component';
+import { AfterViewChecked, Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
 import { FormGroup, FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
-import { NgIf } from '@angular/common';
+import { NgClass, NgFor, NgIf, CommonModule } from '@angular/common';
 import { HttpErrorResponse } from '@angular/common/http';
 import { CaseService } from '../../services/case/case.service';
 import { CaseData } from '../../../app/types/case';
 import { ResultsComponent } from './results/results.component';
-import { CourthouseComponent } from '../common/courthouse/courthouse.component';
+import { initAll } from '@scottish-government/pattern-library/src/all';
+import { Subscription } from 'rxjs';
+import { futureDateValidator } from 'src/app/validators/future-date.validator';
+
+interface ErrorSummaryEntry {
+  fieldId: string;
+  message: string;
+}
+
+const FieldErrors: { [key: string]: { [key: string]: string } } = {
+  case_number: {
+    required: 'You must enter a case number or perform an Advanced search.',
+  },
+  courthouse: {
+    required: 'You must enter a courthouse, if courtroom is filled.',
+  },
+  courtroom: {
+    required:
+      'The courtroom number you have entered is not a recognised number for this courthouse. Check and try again',
+  },
+  date_from: {
+    required: 'You have not selected a start date. Select a start date to define your search',
+    pattern: 'You have not entered a recognised date in the correct format (for example 31/01/2023)',
+    futureDate: 'You have selected a date in the future. The hearing date must be in the past',
+  },
+  date_to: {
+    required: 'You have not selected an end date. Select an end date to define your search',
+    pattern: 'You have not entered a recognised date in the correct format (for example 31/01/2023)',
+    futureDate: 'You have selected a date in the future. The hearing date must be in the past',
+  },
+};
 
 @Component({
   selector: 'app-search',
   templateUrl: './search.component.html',
   styleUrls: ['./search.component.scss'],
   standalone: true,
-  imports: [ReactiveFormsModule, NgIf, ResultsComponent, CourthouseComponent],
+  imports: [CommonModule, ReactiveFormsModule, NgIf, ResultsComponent, NgClass, NgFor, CourthouseComponent],
```

Comment:
If we have `CommonModule` can we remove `NgIf`, `NgClass`, `NgFor`?

Reply:
Yes good shout, would be nice if the linter picked these up :)